packages/@aws-cdk/core/lib/expiration.ts

```diff
@@ -0,0 +1,69 @@
+import { Duration } from './duration';
+/**
+ * Represents a date of expiration.
+ *
+ * The amount can be specified either as a Date object, timestamp, Duration or string.
+ */
+export class Expiration {
+  /**
+   * Expire at the specified date
+   * @param d date to expire at
+   */
+  public static atDate(d: Date) { return new Expiration(d.toUTCString(), d); }
+
+  /**
+   * Expire at the specified timestamp
+   * @param t timestamp in unix milliseconds
+   */
+  public static atTimestamp(t: number) { return Expiration.atDate(new Date(t)); }
+
+  /**
+   * Expire once the specified duration has passed since deployment time
+   * @param t the duration to wait before expiring
+   */
+  public static after(t: Duration) { return Expiration.atDate(new Date(Date.now() + t.toMilliseconds())); }
+
+  /**
+   * Expire at specified date, represented as a string
+   *
+   * @param s the string that represents date to expire at
+   */
+  public static fromString(s: string) { return new Expiration(s, new Date(s)); }
```

Comment:
Why is this needed? `Expiration.atDate(new Date(s))` is short enough, no?

Reply:
It existed in the s3-deployment version and I don't have the domain knowledge to know if this is something that should keep maybe @iliapolo knows more about this

Reply:
From a user's mental model, I think `.date` makes much more sense then `.value`. We can change `s3-deployment` to use `.date.toUTCString()` or whatever it needs.